sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -724,9 +749,32 @@ case class SortMergeJoinExec(
        """.stripMargin
     }
 
+    lazy val semiJoin = {
```

Comment:
How about extracting this block as a private method like `codegenXXXX` just like `HashJoin`?

Reply:
@maropu - yes I was thinking at the first place but worried about number of parameters to be too many. Refined the code a bit and updated now.